File: BACKEND/src/main/java/com/airbnb/clone/Review/service/ReviewService.java
package com.airbnb.clone.Review.service;

import java.util.List;

import com.airbnb.clone.Review.dto.ReviewRequestDto;
import com.airbnb.clone.Review.entity.Review;

public interface ReviewService {

    Review createReview(ReviewRequestDto dto);

    List<Review> getReviewsByPropertyId(Long propertyId);

    List<Review> getReviewsByGuestId(Long guestId);

    boolean hasReviewedBooking(Long bookingId);

    /** Returns {rating: Double, count: long} for a property. rating is null if no reviews. */
    java.util.Map<String, Object> getRatingSummary(Long propertyId);
}

File: FRONTEND/src/services/ReviewService.js
import axios from "axios";

const BASE_URL = "http://localhost:8086/api/v1/reviews";

export const createReview = (reviewData) => {
  return axios.post(BASE_URL, reviewData);
};

export const getReviewsByProperty = (propertyId) => {
  return axios.get(`${BASE_URL}/property/${propertyId}`);
};

export const getReviewsByGuest = (guestId) => {
  return axios.get(`${BASE_URL}/guest/${guestId}`);
};

export const hasReviewedBooking = (bookingId) => {
  return axios.get(`${BASE_URL}/booking/${bookingId}/exists`);
};

/** Returns { rating: number|null, count: number } for a property */
export const getRatingSummary = (propertyId) => {
  return axios.get(`${BASE_URL}/summary/${propertyId}`);
};
